### PR TITLE
improved autocompleteSelect

### PR DESCRIPTION
**Pull request by eek-ai**

### CODE REVIEW

File: src/react-ui-basics/AutocompleteSelect.js [hunk 1]
```diff
@@ -215,11 +215,11 @@ class AutocompleteSelect extends React.Component {
                                    selected={selected}
                                    childComponent={childComponent}
                                    childProps={childProps}
-                                   data={data}
+                                   data={[filterValue, ...data]}
```
Comment (wizzardo) [inline src/react-ui-basics/AutocompleteSelect.js:218]: this should be allowed  only with `allowCustom` flag

File: src/react-ui-basics/AutocompleteSelect.js [hunk 2]
```diff
@@ -333,11 +341,13 @@ class AutocompleteSelect extends React.Component {
...
                         {!withFilter && inputLabel && <label>{inputLabel}</label>}
-                        {(mode === MODE_MULTIPLE_MINI_INLINE || mode === MODE_MULTIPLE || mode === MODE_MULTIPLE_AUTO || mode === MODE_INLINE_MULTIPLE) && list}
+                        {false && (mode === MODE_MULTIPLE_MINI_INLINE || mode === MODE_MULTIPLE || mode === MODE_MULTIPLE_AUTO || mode === MODE_INLINE_MULTIPLE) && list}
                     </div>
                 )}
-                {!(mode === MODE_MULTIPLE_MINI_INLINE || mode === MODE_MULTIPLE || mode === MODE_MULTIPLE_AUTO || mode === MODE_INLINE_MULTIPLE) && list}
+                {false && !(mode === MODE_MULTIPLE_MINI_INLINE || mode === MODE_MULTIPLE || mode === MODE_MULTIPLE_AUTO || mode === MODE_INLINE_MULTIPLE) && list}
             </div>
+              {list}
+            </>
```
Comment (wizzardo) [inline src/react-ui-basics/AutocompleteSelect.js:350]: don't like the idea of rendering `list` outside of the component

File: src/react-ui-basics/AutocompleteSelect.js [hunk 3]
```diff
@@ -307,6 +307,7 @@ class AutocompleteSelect extends React.Component {
                                                   label={inputLabel}
                                                   onChange={this.onChange}
                                                   onFocus={e => this.setState({isActive: true})}
+                                                  onBlur={e => this.setState({isActive: false})}
```
Comment (wizzardo) [inline src/react-ui-basics/AutocompleteSelect.js:310]: this will probably break click in popup

File: src/react-ui-basics/AutocompleteSelect.scss [hunk 4]
```diff
@@ -100,7 +100,7 @@
   }
 
   .TextField {
-    margin-top: 8px;
+    margin-top: 6px;
```
Comment (wizzardo) [inline src/react-ui-basics/AutocompleteSelect.scss:103]: looks not aligned with selected values

File: src/react-ui-basics/AutocompleteSelect.js (hunk 2 above)
Comment (eek-ai) [reply src/react-ui-basics/AutocompleteSelect.js]: moved inside

File: src/react-ui-basics/AutocompleteSelect.js (hunk 1 above)
Comment (eek-ai) [reply src/react-ui-basics/AutocompleteSelect.js]: added condition

**Pull request merged**